pkg/manifests/manifests.go

```diff
@@ -759,6 +765,114 @@ func (f *Factory) KubeStateMetricsDeployment() (*appsv1.Deployment, error) {
 	return d, nil
 }
 
+// KubeStateMetricsDenylistBoundsCheck ensures that the user is:
+// * able to enable metrics that are denied by default, and,
+// * unable to disable metrics that are enabled by default.
+func (f *Factory) KubeStateMetricsDenylistBoundsCheck(deployment *appsv1.Deployment, service *v1.Service, rootCAs *v1.ConfigMap, clientTLS, serverTLS *v1.Secret) (*appsv1.Deployment, error) {
+	// Fail early if the deny-list is empty.
+	if len(f.config.ClusterMonitoringConfiguration.KubeStateMetricsConfig.MetricDenylist) == 0 {
+		return deployment, nil
+	}
+
+	// Fetch the `/metrics` port.
+	var p int32
+	for _, port := range service.Spec.Ports {
+		if port.Name == "https-main" {
+			p = port.Port
+			break
+		}
+	}
+
+	// Query the endpoint.
```

Comment:
Maybe the query part should be refactored into a client method? `KubeStateMetricsDenylistBoundsCheck` is a rather long function and retrieving the metrics endpoint response seems like a good candidate to be split out into its own function?